### PR TITLE
feat(node): Add documentation for `request` param in `httpIntegration` callbacks

### PR DESCRIPTION
**Pull request by Lms24**
This PR adds documentation for the new `request` parameter passed as a second argument to the `ignoreOutgoingRequests` and `ignoreIncomingRequests` callbacks.

ref https://github.com/getsentry/sentry-javascript/issues/12913

### CODE REVIEW

File: docs/platforms/javascript/common/configuration/integrations/http.mdx [hunk 1]
```diff
@@ -45,12 +45,22 @@ If set to false, no breadcrumbs will be captured.
...
 
-_Type: `(url: string) => boolean`_
+_Type: `(url: string, request: RequestOptions) => boolean`_
 
-Allows you to define a method to filter out outgoing requests based on the URL. If the method returns `true`, the request will be ignored.
+Allows you to define a method to filter out outgoing requests based on the URL. If the method returns `true`, no spans or breadcrumbs will be captured for the outgoing request.
+
+The callback function receives two arguments:
+
+- `url`: The full URL of the outgoing request, including the protocol, host, port, path and query string. For example: `https://someService.com/users?name=John`.
```
Comment (andreiborza) [inline docs/platforms/javascript/common/configuration/integrations/http.mdx:54]: l: Ultra nit, but I prefer to use `example.com` for any example urls. Never know what kind of website will live under `someService.com` one day and what user might try open it, even if it's just in docs.

```suggestion
- `url`: The full URL of the outgoing request, including the protocol, host, port, path and query string. For example: `https://example.com/users?name=John`.
```

File: docs/platforms/javascript/common/configuration/integrations/http.mdx [hunk 2]
```diff
@@ -45,12 +45,22 @@ If set to false, no breadcrumbs will be captured.
...
+- `request`: An object of type `RequestOptions` containing the outgoing request's options. You can use this to filter on properties like the the request method or headers.
```
Comment (andreiborza) [inline docs/platforms/javascript/common/configuration/integrations/http.mdx:55]:
```suggestion
- `request`: An object of type `RequestOptions` containing the outgoing request's options. You can use this to filter on properties like the request method or headers.
```

File: docs/platforms/javascript/common/configuration/integrations/http.mdx [hunk 3]
```diff
@@ -45,12 +45,22 @@ If set to false, no breadcrumbs will be captured.
...
-_Type: `(url: string) => boolean`_
+_Type: `(urlPath: string, request: IncomingMessage) => boolean`_
+
+Allows you to define a method to filter out incoming requests based on the URL. If the method returns `true`, no span or transaction will be captured for the incoming request.
+
+The callback function receives two arguments:
 
-Allows you to define a method to filter out incoming requests based on the URL. If the method returns `true`, the request will be ignored.
+- `urlPath`: The URL path of the incoming request, including the query string if available. For example: `/users?name=John`.
+- `request`: An object of type `IncomingMessage` containing the incoming request. You can use this to filter on properties like the the request method or headers.
```
Comment (andreiborza) [inline docs/platforms/javascript/common/configuration/integrations/http.mdx:66]:
```suggestion
- `request`: An object of type `IncomingMessage` containing the incoming request. You can use this to filter on properties like the request method or headers.
```

**Review by andreiborza — APPROVED** [review]

File: docs/platforms/javascript/common/configuration/integrations/http.mdx (hunk 1 above)
Comment (Lms24) [reply docs/platforms/javascript/common/configuration/integrations/http.mdx]: Good point! 😅

**Pull request merged**